EIPS/eip-7867.md

```diff
@@ -0,0 +1,553 @@
+---
+eip: 7867
+title: Flow Control Wallet Call Capability
+description: An EIP-5792 capability providing atomicity and flow control configuration.
+author: Sam Wilson (@SamWilsn) <sam@binarycake.ca>
+discussions-to: https://ethereum-magicians.org/t/wallet-sendcalls-capability-flow-control/22624
+status: Draft
+type: Standards Track
+category: Interface
+created: 2025-01-17
+requires: 5792
+---
+
+## Abstract
+
+This proposal extends [EIP-5792](./eip-5792.md) to allow dapps to downgrade their required atomicity guarantees and control the behaviour after a failed/reverted call. It introduces the batch-scope concept of `strict` vs. `loose` atomicity, where a `strict` batch remains atomic in the face of chain reorgs and a `loose` batch does not; and the per-call ability to continue after a failed/reverted call (`continue`) or stop processing (`halt`).
+
+## Motivation
+
+While the base EIP-5792 specification works extremely well for smart contract wallets, it does not allow the expression of the full range of flow control options that wallets can implement. For example, a dapp may only be submitting a batch for gas savings and not care about whether all calls are reverted on failure. A wallet may only be able to offer a limited form of atomicity through block builder backchannels, but that may be sufficient for a trading platform.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### RPC Interface
+
+The following subsections are modifications to the API endpoints from EIP-5792.
+
+If a request does not match the schema defined below, the wallet MUST reject the request with an error code of 0 <!-- TODO -->.
```

Comment:
i would prefer to create a params/constants table and refer them by their param/constant name here. brings it out as well as confine the TODO to a particualr section.

similarly with other TODOS